### PR TITLE
Uploading media documentation

### PR DESCRIPTION
**Pull request by jimlambie**
This PR adds documentation for uploading media to API

### CODE REVIEW

File: docs/api/latest.md [hunk 1]
```diff
@@ -1553,6 +1553,496 @@ It is possible to supply a custom error message by specifying a `message` proper
...
+      "_version": 1,
+      "_id": "58576e72bafa53b625aebd4f"
+    }
+  ]
+}
+```
+
+#### Filename clashes
+
+If the filename of an file being uploaded is the same as an existing file, the new file will have it's name changed by adding the current timestamp:
```
Comment (eduardoboucas) [inline docs/api/latest.md]: *its*?

File: docs/api/latest.md [hunk 2]
```diff
@@ -1553,6 +1553,496 @@ It is possible to supply a custom error message by specifying a `message` proper
...
+
+* Existing filename: `my-image.jpg`
+* New filename: `test-1480482847099.jpg`
+
+### Referencing files from another collection
+
+Once a file is uploaded, it's identifier can be used to create a reference from another collection. For this example we have a collection called `books` with the following schema:
```
Comment (eduardoboucas) [inline docs/api/latest.md]: *its*?

File: docs/api/latest.md [hunk 3]
```diff
@@ -1553,6 +1553,496 @@ It is possible to supply a custom error message by specifying a `message` proper
...
+    "cache": true,
+    "count": 40,
+    "compose": true,
+    "sort": "title",
+    "sortOrder": 1
+  }
+}
+```
+
+The `image` field is a Reference field which will lookup the default `mediaStore` collection to resolve the reference. Having uploaded an image file and received it's metadata, we can now send a POST to the `books` collection with the image identifier.
```
Comment (eduardoboucas) [inline docs/api/latest.md]: *its*?

File: docs/api/latest.md [hunk 4]
```diff
@@ -1553,6 +1553,496 @@ It is possible to supply a custom error message by specifying a `message` proper
...
+Pre-signed URLs are useful to allow your users or applications to be able to upload a file without requiring an access token. When you request a pre-signed URL, you must provide an access token (see [Authentication](#x)) and specify an expected filename and MIME type for the file to be uploaded. The pre-signed URLs are valid only for the duration specified in the main configuration file or passed in the request for the signed URL.
+
+
+#### Configuration
+
+```json
+"media": {
+  "enabled": true,
+  "tokenSecret": "catbus-goat-omelette",
+  "tokenExpiresIn": "10h"
```
Comment (eduardoboucas) [inline docs/api/latest.md]: I didn't know we supported this human-friendly format for specifying the TTL. We expand on this in the documentation, perhaps?

File: docs/api/latest.md [hunk 5]
```diff
@@ -1553,6 +1553,496 @@ It is possible to supply a custom error message by specifying a `message` proper
...
+Content-Type: application/json
+Authorization: Bearer 8df4a823-1e1e-4bc4-800c-97bb480ccbbe
+
+{
+  fileName: 'my-image.jpg',
+  mimetype: 'image/jpeg'
+}
+```
+
+APO returns a response with a `url` property that contains the signed URL for uploading the specified file:
```
Comment (eduardoboucas) [inline docs/api/latest.md]: *API

Comment (jimlambie) [reply docs/api/latest.md]: FIxed

File: docs/api/latest.md (hunk 3 above)
Comment (jimlambie) [reply docs/api/latest.md]: Fixed

File: docs/api/latest.md (hunk 2 above)
Comment (jimlambie) [reply docs/api/latest.md]: Fixed

File: docs/api/latest.md (hunk 1 above)
Comment (jimlambie) [reply docs/api/latest.md]: Fixed

**Pull request merged**